### PR TITLE
add DDP

### PR DESCRIPTION
**Pull request by LongaBonga**

### CODE REVIEW

File: scripts/main.py [hunk 1]
```diff
@@ -58,7 +61,7 @@ def main():
 
     cifar100_train = datasets.CIFAR100('path/to/cifar100_root/', train = True, download=True, transform= train_transform)
     train_data = torch.utils.data.DataLoader(cifar100_train,
```
Comment (sovrasov) [inline scripts/main.py:63]: Should we also use DistributedSampler?

File: scripts/train.py [hunk 2]
```diff
@@ -53,10 +40,10 @@ def train_func(args, model, criterion, optimizer, train_dataloader, test_dataloa
         val_loss, val_pred, val_size = val_func(model, criterion, optimizer, test_dataloader, device, writer, epoch)
 
 
-        print_at_master('Train loss:' + str(train_loss / train_size))
-        print_at_master('Val loss:' + str(val_loss / val_size))
-        print_at_master('Train acc:' + str((train_pred / train_size)*100))
-        print_at_master('Val acc:' +str((val_pred / val_size)*100))
+        print_at_master('Train loss: {}'.format((train_loss / train_size)))
```
Comment (sovrasov) [inline scripts/train.py:43]: `f'Train loss: {train_loss / train_size})'`

File: scripts/main.py [hunk 3]
```diff
@@ -25,65 +26,36 @@ def main():
     parser.add_argument('--output_dir', type=str, default='', help='directory to store training artifacts')
     parser.add_argument('--model', type=str, default='mobilenet_v2', help='name of model')
     parser.add_argument('--optimazer', type=str, default='sgd', help='name of optimazer')
+    parser.add_argument('--batch_size', type=int, default='64', help='batch size')
     parser.add_argument('--mode', type=str, default='train', help='mode of model train/evaluation')
     parser.add_argument('--device', type=str, default='cpu', choices=['cuda','cpu'],
                         help='choose device to train on')
+    parser.add_argument("--local_rank", default=0, type=int)
+    parser.add_argument("--fp16", default=True, type=bool, choices=[True, False])
```
Comment (sovrasov) [inline scripts/main.py:34]: Consider action='store_true'. It's more convenient

Comment (sovrasov) [reply scripts/main.py]: https://stackoverflow.com/questions/8203622/argparse-store-false-if-unspecified

File: .gitignore [hunk 4]
```diff
@@ -4,3 +4,5 @@ __pycache__/
 /scripts/runs/
 .ipynb_checkpoints/
 *.pth
+torch==1.7.1
```
Comment (sovrasov) [inline .gitignore:7]: Why is this here?

File: requirements.txt [hunk 5]
```diff
@@ -0,0 +1,3 @@
+torch==1.7.1
```
Comment (sovrasov) [inline requirements.txt:1]: Why do you need to use a specific pytorch version here?

File: scripts/builders/optim_builder.py [hunk 6]
```diff
@@ -5,12 +5,12 @@
 def build_optimizer(net, optimizer_name):
     assert optimizer_name in AVAILABLE_OPTIMS
     if optimizer_name == 'adadelta':
-        optim = torch.optim.Adadelta(net.parameters(), lr=5e-4)
+        optim = torch.optim.Adadelta(net.parameters(), lr=1e-3)
```
Comment (sovrasov) [inline scripts/builders/optim_builder.py:8]: I propose make LR adjustable from the command line

File: scripts/data_loading/data_loader.py [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+    ])
+
+
+    sampler_train = None
+    sampler_val = None
+
+    data_path_train = os.path.join(args.data_path, 'train')
+    data_path_val = os.path.join(args.data_path, 'val')
+
+    train_dataset = datasets.CIFAR100(args.data_path, transform= train_transform)
```
Comment (sovrasov) [inline scripts/data_loading/data_loader.py:29]: Use ImageFolder here

File: scripts/data_loading/data_loader.py [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+    transforms.ToTensor(),
+    transforms.Normalize(mean=[0.485, 0.456, 0.406],
+                        std=[0.229, 0.224, 0.225]),
+    ])
+
+
+    sampler_train = None
+    sampler_val = None
+
+    data_path_train = os.path.join(args.data_path, 'train')
```
Comment (sovrasov) [inline scripts/data_loading/data_loader.py:26]: Unused variables

File: scripts/data_loading/data_loader.py [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+
+    if num_distrib() > 1:
+        sampler_train = torch.utils.data.distributed.DistributedSampler(train_dataset)
+        sampler_val = torch.utils.data.distributed.DistributedSampler(val_dataset) 
+
+    train_data = torch.utils.data.DataLoader(train_dataset,
+                                            batch_size=args.batch_size,
+                                            shuffle=False,
+                                            num_workers=4,
+                                            sampler=sampler_train)
```
Comment (sovrasov) [inline scripts/data_loading/data_loader.py:40]: What if distributed training is off and sampler_train is not defined?
